ledger/state_test.go

```diff
@@ -0,0 +1,464 @@
+			t.Fatalf("failed to add test block: %v", err)
+		}
+	}
+
+	// Create LedgerState
+	ls := &LedgerState{
+		chain: primaryChain,
+		db:    db,
+	}
+
+	// Test retrieving each block
+	for i, expectedBlock := range testBlocks {
+		point := ocommon.NewPoint(expectedBlock.Slot, expectedBlock.Hash)
+		result, err := ls.GetBlock(point)
+		if err != nil {
+			t.Errorf("GetBlock failed for block %d: %v", i, err)
+			continue
+		}
+
+		if result.Slot != expectedBlock.Slot {
+			t.Errorf("Block %d: expected slot %d, got %d", i, expectedBlock.Slot, result.Slot)
+		}
+		if string(result.Hash) != string(expectedBlock.Hash) {
+			t.Errorf("Block %d: expected hash %x, got %x", i, expectedBlock.Hash, result.Hash)
+		}
+		if result.Number != expectedBlock.Number {
+			t.Errorf("Block %d: expected number %d, got %d", i, expectedBlock.Number, result.Number)
+		}
+	}
+}
+
+// TestGetBlock_ErrorPropagation tests that errors from chain are properly propagated
+func TestGetBlock_ErrorPropagation(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create LedgerState
+	ls := &LedgerState{
+		chain: cm.PrimaryChain(),
+		db:    db,
+	}
+
+	// Test with various invalid points
+	testCases := []struct {
+		name string
+		point ocommon.Point
+	}{
+		{
+			name: "non-existent slot",
+			point: ocommon.NewPoint(999999999, []byte{0xde, 0xad, 0xbe, 0xef}),
+		},
+		{
+			name: "wrong hash for slot",
+			point: ocommon.NewPoint(100, []byte{0x00, 0x00, 0x00, 0x00}),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := ls.GetBlock(tc.point)
+			if err == nil {
+				t.Error("Expected error, got nil")
+			}
+			// Verify zero value is returned
+			if result.ID != 0 || len(result.Hash) != 0 {
+				t.Error("Expected zero value on error")
+			}
+		})
+	}
+}
+
+// TestGetBlock_ConsistentReturns tests that repeated calls return consistent results
+func TestGetBlock_ConsistentReturns(t *testing.T) {
+	// Setup test database
+	db, err := database.New(&database.Config{
+		BlobCacheSize: 1 << 20,
+		DataDir:       "",
+	})
+	if err != nil {
+		t.Fatalf("failed to create test database: %v", err)
+	}
+	defer db.Close()
+
+	// Create chain manager
+	cm, err := chain.NewManager(db, nil)
+	if err != nil {
+		t.Fatalf("failed to create chain manager: %v", err)
+	}
+
+	// Create test block
+	testBlock := models.Block{
+		Hash:     []byte{0xde, 0xad, 0xbe, 0xef},
+		Slot:     777,
+		Number:   77,
+		PrevHash: []byte{0x00, 0x00, 0x00, 0x00},
+		Cbor:     []byte{0x82, 0x00, 0x81, 0x00},
+		Type:     5,
+	}
+
+	primaryChain := cm.PrimaryChain()
+	if err := primaryChain.AddBlock(testBlock, nil); err != nil {
```

Comment:
@coderabbitai This should be a "github.com/blinklabs-io/gouroboros/ledger/common".Block not a models.Block struct. You can use Decode() on models.Block to get a ledger.Block suitable for use here.